driver-core/src/test/functional/com/mongodb/JsonTestServerVersionChecker.java

```diff
@@ -51,9 +55,8 @@ private static boolean canRunTest(final BsonDocument document, final ServerVersi
                 && serverVersion.compareTo(getMaxServerVersionForField("maxServerVersion", document)) > 0) {
             return false;
         }
-        if (document.containsKey("topology")) {
-            BsonArray topologyTypes = document.getArray("topology");
-            return topologyMatches(topologyTypes);
```

Comment:
This was a bug.  We only on want to return if we know we're skipping.  Otherwise we don't execute any of the checks below, one of which may be false.